### DISCUSSION TITLE
请问有没有办法动态获取jsmind的高度

### DISCUSSION THREAD

**Discussion by icefire1982** [hizzgdev/jsmind#550] [General]
目前的方案中，先是制定container的固定高度，然后生成jsmind思维导图，但是就会遇到一个问题：有时思维导图的高度比container的高度大不少，导致与页面其他内容间隔很大；有时思维导图的高度又超过container的高度，导致出现滚动条。有没有什么办法，可以在生成思维导图后获取jsmind的高度，然后通过javascript来动态调整container的高度，使其即可在高度方面能全部呈现思维导图，又不至于会出现与其他元素间距过大或者不得不显示滚动条的情况。

**Comment by hizzgdev** [comment]
我把问题移动到了 Discussion 区，它并不是一个 Issue 哈。 @icefire1982 

你可以读取 container 里的子元素的 `offsetWidth` 和 `offsetHeight` 。子元素应该只有一个，所以应该不难办到：

```
<div class="jsmind-inner jmnode-overflow-hidden">...</div>
```

不过提醒一下哈，脑图有时候会很大，有时会比较小，如果你想让它“自由生长”的话，可能要小心不要破坏了页面的布局。
